### PR TITLE
Added in Tktable.mk

### PR DESCRIPTION
**Pull request by highperformancecoder**
Please read http://mxe.cc/#creating-packages

In particular, make sure that your build rules:
- install .pc file,
- install bin/test-pkg.exe compiled with flags by pkg-config,
- install .dll to bin/ and .a, .dll.a to lib/,
- use $(TARGET)-cmake instead of cmake,
- build in `$(BUILD_DIR)` instead of `$(SOURCE_DIR)`,
- do not run target executables with Wine,
- do not download anything while building,
- do not install documentation,
- do not install .exe files except test and build systems,

and .patch files are generated by tools/patch-tool-mxe.

If you add a package, you can use tool tools/skeleton.py.

Thanks!

### CODE REVIEW

File: plugins/tcl.tk/TkTable.mk [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+define $(PKG)_UPDATE
+    $(WGET) -q -O- 'https://sourceforge.net/projects/minsky/files/Sources/' | \
+    $(SED) -n 's,.*TkTable\.\([0-9][^>]*\)\.tar.*,\1,p' | \
+    head -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && ./configure \
+        $(MXE_CONFIGURE_OPTS) \
+        --disable-shared \
```
Comment (starius) [inline plugins/tcl.tk/TkTable.mk]: `MXE_CONFIGURE_OPTS` already does `--disable-shared` for static builds.
By the way, does shared build work?

File: plugins/tcl.tk/TkTable.mk [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# This file is part of MXE.
+# See index.html for further information.
+
+PKG             := TkTable
```
Comment (starius) [inline plugins/tcl.tk/TkTable.mk]: We do not use uppercase characters in package names. Can you change the name to tktable, please?

File: plugins/tcl.tk/TkTable.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    head -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && ./configure \
+        $(MXE_CONFIGURE_OPTS) \
+        --disable-shared \
+        --with-tcl=$(PREFIX)/$(TARGET)/lib --with-tk=$(PREFIX)/$(TARGET)/lib
+    $(MAKE) -C '$(1)' -j '$(JOBS)'
+    cp $(1)/libTktable2.11.a $(PREFIX)/$(TARGET)/lib
```
Comment (starius) [inline plugins/tcl.tk/TkTable.mk]: Can the package be installed with `make install` rather than `cp`?

File: plugins/tcl.tk/TkTable.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# patches.
+
+define $(PKG)_UPDATE
+    $(WGET) -q -O- 'https://sourceforge.net/projects/minsky/files/Sources/' | \
+    $(SED) -n 's,.*TkTable\.\([0-9][^>]*\)\.tar.*,\1,p' | \
+    head -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && ./configure \
```
Comment (starius) [inline plugins/tcl.tk/TkTable.mk]: Does out-of-source build work?
Try
```
cd '$(BUILD_DIR)' && $(SOURCE_DIR)/configure \
...
$(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
```

**Comment by highperformancecoder** [comment]
Thanks Boris, I'll check whether the dynamic build works. I'll also
change the package name.

Also - re make install, the default make install target for TkTable puts
the library in a weird location. I'll see if that can be overridden
with a --prefix configure option, but I'm not hopeful.

Cheers, Russell


-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders
Visiting Senior Research Fellow        hpcoder@hpcoders.com.au
Economics, Kingston University         http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by starius** [comment]
By the way, `MXE_CONFIGURE_OPTS` sets prefix as well. Make sure to use only one of `MXE_CONFIGURE_OPTS` and `--prefix`.

**Comment by highperformancecoder** [comment]
Hi Starius - sorry for the long wait. I've just checked in a change addressing your concerns.

File: plugins/tcl.tk/tktable.mk [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+define $(PKG)_UPDATE
+    $(WGET) -q -O- 'https://sourceforge.net/projects/minsky/files/Sources/' | \
+    $(SED) -n 's,.*TkTable\.\([0-9][^>]*\)\.tar.*,\1,p' | \
+    head -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && ./configure \
+        $(MXE_CONFIGURE_OPTS) \
+	--without-x \
```
Comment (starius) [inline plugins/tcl.tk/tktable.mk]: Fix indentation.

File: plugins/tcl.tk/tktable.mk [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        --with-tcl=$(PREFIX)/$(TARGET)/lib --with-tk=$(PREFIX)/$(TARGET)/lib
+# bizarrely, the Makefile links against -lX11 for no reason, even if --without-x is specified
```
Comment (starius) [inline plugins/tcl.tk/tktable.mk]: Fix indentation.

File: plugins/tcl.tk/TkTable.mk (hunk 2 above)
Comment (highperformancecoder) [reply plugins/tcl.tk/TkTable.mk]: done

File: plugins/tcl.tk/TkTable.mk (hunk 1 above)
Comment (highperformancecoder) [reply plugins/tcl.tk/TkTable.mk]: Fixed

File: plugins/tcl.tk/TkTable.mk (hunk 3 above)
Comment (highperformancecoder) [reply plugins/tcl.tk/TkTable.mk]: Done

File: plugins/tcl.tk/TkTable.mk (hunk 4 above)
Comment (highperformancecoder) [reply plugins/tcl.tk/TkTable.mk]: make and make install are used now, with some hacking of variables passed to make.

**Comment by tonytheodore** [comment]
Shared build on darwin is broken https://gist.github.com/tonytheodore/edb28ffa8ee5337ff68541f72e5e2620, it appears to be picking up the wrong platform flags.

There's some new style changes for out-of-source builds:
```diff
diff --git a/plugins/tcl.tk/tktable.mk b/plugins/tcl.tk/tktable.mk
index b5560bf0..ef598648 100644
--- a/plugins/tcl.tk/tktable.mk
+++ b/plugins/tcl.tk/tktable.mk
@@ -1,5 +1,4 @@
-# This file is part of MXE.
-# See index.html for further information.
+# This file is part of MXE. See LICENSE.md for licensing information.

 PKG             := tktable
 $(PKG)_OWNER    := https://github.com/highperformancecoder
@@ -26,12 +25,13 @@ define $(PKG)_UPDATE
 endef

 define $(PKG)_BUILD
-    cd '$(1)' && ./configure \
+    cd '$(BUILD_DIR)' && '$(SOURCE_DIR)/configure' \
         $(MXE_CONFIGURE_OPTS) \
         --without-x \
-        --with-tcl=$(PREFIX)/$(TARGET)/lib --with-tk=$(PREFIX)/$(TARGET)/lib
+        --with-tcl=$(PREFIX)/$(TARGET)/lib \
+        --with-tk=$(PREFIX)/$(TARGET)/lib
     # bizarrely, the Makefile links against -lX11 for no reason, even if
     # --without-x is specified
-    $(MAKE) -C '$(1)' -j '$(JOBS)' LIBS=
-    $(MAKE) -C '$(1)'  PKG_DIR= install
+    $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)' LIBS=
+    $(MAKE) -C '$(BUILD_DIR)' -j 1 install PKG_DIR=
 endef
```

can you apply those and rebase into a single commit?

**Comment by highperformancecoder** [comment]
Thanks Tony. Done that now, and learnt something about squashing commits in the process. It took me 5 goes to get it right, even with a StackOverflow recipe!

**Comment by tonytheodore** [comment]
You seem to have missed:
```
cd '$(BUILD_DIR)' && '$(SOURCE_DIR)/configure' \
```

**Comment by highperformancecoder** [comment]
Hi Tony, you must be misreading the diff. All occurrances of $(1) have been replace by $(BUILD_DIR).
Cheers

**Comment by tonytheodore** [comment]
There's no `./configure` in `$(BUILD_DIR)`?

**Comment by highperformancecoder** [comment]
Sorry,I misunderstood what BUILD_DIR was. I thought it was the directory where the package was unpacked and built. But I see now that the intent is to separate the two directories (for whatever reason).

That became clear when my test build finished (well after your review :).

**Comment by tonytheodore** [comment]
Separating BUILD_DIR lets you test against a development repo with:
```
make tktable_SOURCE_TREE=~/path/to/local/sources
```
without having to create a tarball etc.

Thanks!

**Pull request merged**